### PR TITLE
Reworked some parts of FOLF

### PR DESCRIPTION
**Pull request by MCWertGaming**
A small rework of some parts of FOLF. I moved the example programs back to FOLF and reworked the cmake system. Now everything is handled by cmake itself for better compatibility. Some features and fixes are included as well.
- moved example programs back
- reworked the cmake system
- removed some old code that is not used
- added headers to all files with better copyright notice and description
- reworked the waitFor function to remove unececessary includes
- added the timer class by SNEK

### CODE REVIEW

File: folf-namespace/folf.hpp [hunk 1]
```diff
@@ -1,19 +1,14 @@
...
 #pragma once
+
 // includes various functions for benchmarking and common tasks
 namespace folf 
 {
-    namespace aboutFolf 
-    {
-        const char *folfVersion();
-    }
     namespace numberOperations
```
Comment (AdmiralEmser) [inline folf-namespace/folf.hpp:12]: e.g. timeOperations more understandable?

File: folf-namespace/numberOperations.cpp [hunk 2]
```diff
@@ -1,18 +1,23 @@
...
 #include <cmath>
 #include <random>
 #include <chrono>
 
 bool folf::numberOperations::checkPrime(const unsigned long long *num)
 {
     // taking the square root to speed up the calculation and place the result into another variable to save more time (seems like c++ calculates the sqrt on every loop otherwise)
-    double root = sqrt(*num);
-    for (int j = 3; j <= root; j += 2) 
+    double root = sqrt((double)*num);
```
Comment (AdmiralEmser) [inline folf-namespace/numberOperations.cpp:16]: unsigned double?

Comment (MCWertGaming) [reply folf-namespace/numberOperations.cpp:16]: you can't make a double unsigned

**Review by AdmiralEmser — APPROVED** [review]

**Comment by AdmiralEmser** [comment]
Looks great :)

**Pull request merged**